v0/destinations/hs/transform.js

```diff
@@ -172,6 +175,41 @@ async function processSingleMessage(message, destination) {
 }
 
 function process(event) {
-  return processSingleMessage(event.message, event.destination);
+  return event.message;
 }
-exports.process = process;
+
+const formatBatchResponse = (batchPayload, metadataList, destination) => {
+  const response = defaultBatchRequestConfig();
+  response.batchedRequest = batchPayload;
+  response.metadata = metadataList;
+  response.destination = destination;
+  return response;
+};
+
+const batch = async destEvents => {
+  const respList = [];
+  for (let index = 0; index < destEvents.length; index += 1) {
+    const ev = destEvents[index];
+    const { message, metadata, destination } = ev;
+    if (message.statusCode) {
+      // processed event
+      respList.push(formatBatchResponse(message, [metadata], destination));
+    } else {
+      // unprocessed event
+      try {
+        const singleResponse = await processSingleMessage(message, destination);
+        respList.push(
+          formatBatchResponse(singleResponse, [metadata], destination)
+        );
+      } catch (error) {
+        logger.debug(error);
```

Comment:
Earlier, failed transformations were caught in `proc_errors` and we can notify customers. 

With batch API and failed transformation, we are losing the error message. We are silently dropping the error message here. Not easy to associate this with jobs.

Router is treating this as 500 and retries for 3 hours. So all invalid events will stay in our system for 3 hours.

We missed this while designing batch API earlier. Something we should fix.

Reply:
CC: @chandumlg